components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/model/template/service/BallerinaPath.java

```diff
@@ -41,51 +42,87 @@ public BallerinaPath() {
+            try {
+                operation = new BallerinaOperation().buildContext(item.getPut(), api);
+            } catch (CLICompileTimeException e) {
+                throw new CLICompileTimeException("Error while parsing the information under PUT resource.\n\t-"
+                        + e.getErrorMessage());
+            }
             entry = new AbstractMap.SimpleEntry<>("put", operation);
             operations.add(entry);
         }
         if (item.getPost() != null) {
             setServersToOperationLevel(item.getPost(), item.getServers());
-            operation = new BallerinaOperation().buildContext(item.getPost(), api);
+            try {
+                operation = new BallerinaOperation().buildContext(item.getPost(), api);
+            } catch (CLICompileTimeException e) {
+                throw new CLICompileTimeException("Error while parsing the information under POST resource.\n\t-"
+                        + e.getErrorMessage());
+            }
             entry = new AbstractMap.SimpleEntry<>("post", operation);
             operations.add(entry);
         }
         if (item.getDelete() != null) {
             setServersToOperationLevel(item.getDelete(), item.getServers());
-            operation = new BallerinaOperation().buildContext(item.getDelete(), api);
+            try {
+                operation = new BallerinaOperation().buildContext(item.getDelete(), api);
+            } catch (CLICompileTimeException e) {
+                throw new CLICompileTimeException("Error while parsing the information under DELETE resource.\n\t-"
+                        + e.getErrorMessage());
+            }
             entry = new AbstractMap.SimpleEntry<>("delete", operation);
             operations.add(entry);
         }
         if (item.getOptions() != null) {
             setServersToOperationLevel(item.getOptions(), item.getServers());
-            operation = new BallerinaOperation().buildContext(item.getOptions(), api);
+            try {
+                operation = new BallerinaOperation().buildContext(item.getOptions(), api);
+            } catch (CLICompileTimeException e) {
+                throw new CLICompileTimeException("Error while parsing the information under OPTIONS resource.\n\t-"
+                        + e.getErrorMessage());
+            }
             entry = new AbstractMap.SimpleEntry<>("options", operation);
             operations.add(entry);
         }
         if (item.getHead() != null) {
             setServersToOperationLevel(item.getHead(), item.getServers());
-            operation = new BallerinaOperation().buildContext(item.getHead(), api);
+            try {
+                operation = new BallerinaOperation().buildContext(item.getHead(), api);
+            } catch (CLICompileTimeException e) {
+                throw new CLICompileTimeException("Error while parsing the information under HEAD resource.\n\t-"
```

Comment:
Shouldn't we propagate e to the top? not only the message?

Reply:
Yes its better to propagate the stack trace to the top. So the stack trace will be in the log file instead of the console

Reply:
Yes. I will fix that.

Reply:
fixed via https://github.com/wso2/product-microgateway/pull/1072/commits/8b13b9b1338bc7afc348bf0799d3db5050254d2d